addons/point_of_sale/static/src/app/store/pos_store.js

```diff
@@ -1049,7 +1049,9 @@ export class PosStore extends Reactive {
                 // Replace the original session by the rescue one. And the rescue one will have
                 // a higher id than the original one since it's the last one created.
                 const session = this.models["pos.session"].sort((a, b) => a.id - b.id)[0];
-                session.delete();
+                if (session.state !== "opened") {
+                    session.delete();
+                }
```

Comment:
My apologies for the time I took for the review.
 
Although we don't have the traceback anymore with your fix, I'm not convinced this is the direction we should follow. If I understand the meaning of the comment above, it seems like the session is supposed to be deleted all the time and not conditionally.

To me, the biggest problem (before your fix) is the fact that after the session is deleted, the orders are not linked to any session. And again, reading the comment, it seems to me that we should re-link all orders to the rescue session (the one not deleted). Maybe we should do this here?

I tested the idea locally in an environment with only 1 order and it seems to work. Here's the quick test that I did:
```js
                if (modelToAdd["pos.session"].length > 0) {
                    const session = this.models["pos.session"].sort((a,b)=>a.id - b.id)[0];
                    session.delete();
                    this.models["pos.order"].getAll()[0].session_id = this.models["pos.session"].sort((a,b)=>a.id - b.id)[0]
                }
```
Obviously this has to be adapted in case we have multiple orders, and it has to be cleaner, but it should also solve the problem. 

Would you mind trying this instead?

Reply:
> To me, the biggest problem (before your fix) is the fact that after the session is deleted, the orders are not linked to any session. And again, reading the comment, it seems to me that we should re-link all orders to the rescue session (the one not deleted). Maybe we should do this here?

Yes, you are correct. 
I believe this might be the reason why we are not obtaining the session ID for orders.

Thanks for suggestion @sbel-odoo 🙇🏻